src/Cassandra/Connections/Connection.cs

```diff
@@ -536,18 +537,21 @@ internal bool ReadParse(byte[] buffer, int length)
 
             // Check if protocol version has already been determined (first message)
             ProtocolVersion protocolVersion;
-            var headerLength = Volatile.Read(ref _frameHeaderSize);
+            var headerLength = _frameHeaderSize;
```

Comment:
hm... I think we still need the volatile read. The memory barrier below prevents reordering on that case, but `_frameHeaderSize` may still hold an old value for a given thread.

Reply:
Yeah I was trying to avoid adding more volatile operations to this path but I don't think we have a choice. Same for the `serializer` instance